tests/src/test/scala/whisk/core/containerpool/test/ContainerPoolTests.scala

```diff
@@ -415,18 +573,18 @@ class ContainerPoolObjectTests extends FlatSpec with Matchers with MockFactory {
   behavior of "ContainerPool remove()"
 
   it should "not provide a container if pool is empty" in {
-    ContainerPool.remove(Map()) shouldBe None
+    ContainerPool.remove(Map(), MemoryLimit.stdMemory) shouldBe List.empty
   }
 
   it should "not provide a container from busy pool with non-warm containers" in {
     val pool = Map('none -> noData(), 'pre -> preWarmedData())
-    ContainerPool.remove(pool) shouldBe None
+    ContainerPool.remove(pool, MemoryLimit.stdMemory) shouldBe List.empty
```

Comment:
Would it make sense to add a test which tests that if you cannot make space for enough capacity, the List is empty?